Sources/AppBundle/ui/SecureInputView.swift

```diff
@@ -0,0 +1,93 @@
+import AppKit
+import Carbon
+import SwiftUI
+
+private let iconSize = CGSize(width: 50, height: 50)
+private let textSize = CGSize(width: 440, height: 100)
+
+public class SecureInputPanel: NSPanel {
+    @MainActor public static var shared: SecureInputPanel = SecureInputPanel()
+    private var hostingView = NSHostingView(rootView: SecureInputView())
+
+    private init() {
+        super.init(
+            contentRect: .zero,
+            styleMask: [.nonactivatingPanel, .borderless, .hudWindow, .utilityWindow],
+            backing: .buffered,
+            defer: false,
+        )
+        self.level = .floating
+        self.collectionBehavior = [.canJoinAllSpaces, .fullScreenAuxiliary]
+        self.isReleasedWhenClosed = false
+        self.hidesOnDeactivate = false
+        self.isMovableByWindowBackground = false
+        self.alphaValue = 1
+        self.hasShadow = true
+        self.backgroundColor = .clear
+    }
+
+    public func refresh() {
+        self.contentView?.subviews.removeAll()
+        if IsSecureEventInputEnabled() {
+            hostingView = NSHostingView(rootView: SecureInputView())
+            hostingView.frame = NSRect(x: 0, y: 0, width: iconSize.width, height: iconSize.height)
+            self.contentView?.addSubview(hostingView)
+            let x = mainMonitor.width - iconSize.width - 20
+            let panelFrame = NSRect(x: x, y: 20, width: iconSize.width, height: iconSize.width)
+            self.setFrame(panelFrame, display: true)
+            self.orderFrontRegardless()
+        } else {
+            close()
+        }
+    }
+
+    public func updateFrame(isMinimized: Bool) {
+        let width = isMinimized ? iconSize.width : textSize.width
+        let height = isMinimized ? iconSize.height : textSize.height
+        hostingView.frame = NSRect(x: 0, y: 0, width: width, height: height)
+        let x = mainMonitor.width - width - 20
+        let panelFrame = NSRect(x: x, y: 20, width: width, height: height)
+        self.setFrame(panelFrame, display: true)
+    }
+}
+
+struct SecureInputView: View {
+    @State var isMinimized: Bool = true
+
+    @Environment(\.colorScheme) var colorScheme: ColorScheme
+    private var fontColor: Color {
+        colorScheme == .dark ? Color.black : Color.white
+    }
+
+    var body: some View {
+        ZStack(alignment: .center) {
+            Rectangle()
+                .fill(Color.gray.opacity(isMinimized ? 0.8 : 1.0))
+            if isMinimized {
+                Image(systemName: "lock.shield.fill")
+                    .resizable()
+                    .aspectRatio(contentMode: .fit)
+                    .padding(6)
+            } else {
+                Text("**Secure Input** is active. **Secure Input** is a macOS security feature that prevents applications from reading keyboard events.")
```

Comment:
```suggestion
                Text("AeroSpace cannot respond to keyboard shortcuts while **Secure Input** is active. **Secure Input** is a macOS security feature that prevents applications from reading keyboard events.")
```

We should make it clear that the message comes from AeroSpace